lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -2132,6 +2112,35 @@ def each_host_range_chunk(host_ranges, &block)
 
   #######
   private
+
+  def run_nmap(arguments, nmap, use_sudo=false)
+    begin
+      nmap_pipe = use_sudo ? ::Open3::popen3('sudo', nmap, *arguments) : ::Open3::popen3([nmap, 'nmap'], *arguments)
+      temp_nmap_threads = []
+      temp_nmap_threads << framework.threads.spawn("db_nmap-Stdout", false, nmap_pipe[1]) do |np_1|
+        np_1.each_line do |nmap_out|
+          next if nmap_out.strip.empty?
+          print_status("Nmap: #{nmap_out.strip}")
+        end
+      end
+
+      temp_nmap_threads << framework.threads.spawn("db_nmap-Stderr", false, nmap_pipe[2]) do |np_2|
+
+        np_2.each_line do |nmap_err|
+          next if nmap_err.strip.empty?
+          print_status("Nmap: '#{nmap_err.strip}'")
+          if nmap_err.strip == 'You requested a scan type which requires root privileges.'
+            return run_nmap(arguments, nmap, true)
```

Comment:
```suggestion
            return run_nmap(nmap_path, arguments, use_sudo: true)
```

Reply:
nab: I'm not a _massive_ fan of the recursion here 🤔

Reply:
> nab: I'm not a massive fan of the recursion here thinking

yea I don't love it, but it was the easiest way to get it done, I couldn't think of another way without doing a bunch of refactoring that I wanted to avoid

Reply:
We'd also want to guard clause this to only run in environments that support sudo; I'm not sure what the behavior here is here for cygwin/windows

Reply:
![image](https://user-images.githubusercontent.com/19910435/103651597-c7574100-4f59-11eb-811e-683b560323f7.png)
so on windows it's not an issue, let me check cygwin real quick too

Reply:
also looks like it's only possible to run nmap as an administrator on windows so we shouldn't see any problems there